src/Microsoft.AspNet.Mvc.Razor/Compilation/SyntaxTreeGenerator.cs

```diff
@@ -17,6 +17,19 @@ public static class SyntaxTreeGenerator
             string path,
             CompilationSettings compilationSettings)
```

Comment:
Looking at this now, it doesn't really make sense to have a method that doesn't do any compilation take in compilation settings. It'd probably be best removing this overload and just rely on the parseoptions variations.